### DISCUSSION TITLE
r2d2 connection pool causing app to quit unexpectedly

### DISCUSSION THREAD

**Discussion by ohuu** [diesel-rs/diesel#3444] [Q&A] [answered]
I'm fairly new to Rust and I've been trying to recreate an app that I wrote in Scala in Rust to see how it goes. So far so good except when I try to create a connection pool using r2d2. Sometimes (not always) the app unexpectedly quits when creating the pool. Here is my code:

```rust
use actix_web::{web, App, HttpServer};
use diesel::pg::PgConnection;
use diesel::r2d2::{ConnectionManager, Pool};
use routes::hello;
use std::env;

#[tokio::main]
async fn main() -> std::io::Result<()> {
    let db_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
    let conn_manager = ConnectionManager::<PgConnection>::new(db_url);
    let pool = Pool::builder()
        .build(conn_manager)
        .expect("Failed to create pool.");

    HttpServer::new(move || {
        println!("new worker");
        App::new()
            .app_data(web::Data::new(pool.clone()))
            .route("/", web::get().to(hello))
    })
    .bind(("127.0.0.1", 8080))?
    .run()
    .await
}
```

Nine times out of ten it fails with this error `error: process didn't exit successfully: 'target\debug\accuheat_rust.exe\' (exit code: 3)`. No other information is given, no panic or anything.

My database is running inside a docker container, it has 100 max connections and I can connect to it just fine. I've got no more information to go on, else I would probably be able to fix this issue so I'm hoping someone else has experienced this problem before.

I'm running Windows and I'm using the msvc version of the rust toolchain.

**Comment by weiznich** [comment, marked as answer]
That's very likely a duplicate of https://github.com/diesel-rs/diesel/discussions/2947

The issue there is an bug in your libpq distribution. You can either fix that by replacing some internal DLL's there or by using a libpq version from another source.

**Reply by ohuu** [reply]
thanks for the super quick reply. It turns out that is exactly what the problem was! replacing libintl-9.dll did the trick. Hopefully EnterpriseDB will fix this issue soon. It hasn't been fixed in version 15.1 so I'll give them a nudge. Thanks again!
